### PR TITLE
Updated and Added a roof to home improvement superstore

### PR DESCRIPTION
**Pull request by FarkasKBence**
#### Summary

 ```SUMMARY: Content "Updated and Added a roof to home improvement superstore"```

#### Purpose of change
The home improvement superstore is a very old building, so I updated it.

#### Describe the solution
*  Added a roof with air conditioners, skylights, water tanks, etc.
* Added new furnitures, such as dishwashers, wardrobes, mirrors, filing cabinets, a piano and a water heater.
* New terrains, such as golf grass and woodchips.
* Added more items, like carpets and more raw materials.
* Cement and sand now spawn in canvas sacks.
* Added a bigger parking lot

#### Screenshots

homeimprovementsuperstore1

homeimprovemtsuperstore

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
```SUMMARY Content "Updated and Added a roof to home improvement superstore"```

**Comment by anothersimulacrum** [comment]
Jenkins rebuild

**Comment by Vasyan2006** [comment]
No roof on level 2 to cover stairs?

**Comment by LovamKicsiGazsii** [comment]
> No roof on level 2 to cover stairs?

It has one.
proof
There is no need to do this part manually.

File: data/json/overmap_terrain.json [hunk 1]
```diff
@@ -2551,14 +2551,68 @@
...
+    "extras": "field"
+  },
+  {
+    "type": "overmap_terrain",
+    "id": "hdwr_large_0_1_1",
+    "name": "home improvement superstore",
+    "sym": 46,
+    "color": "light_blue",
+    "see_cost": 5,
+    "extras": "field"
```
Comment (ZhilkinSerg) [inline data/json/overmap_terrain.json]:
```suggestion
    "extras": "field"
```

File: data/json/mapgen/homeimprovement_superstore.json [hunk 2]
```diff
@@ -90,478 +95,344 @@
...
+        "V": { "item": "vending_drink_home_impr", "chance": 85, "repeat": [ 5, 8 ] },
+        "1": { "item": "paint", "chance": 65 },
+        "2": { "item": "carpet", "chance": 65 },
+        "3": [ { "item": "farming_seeds", "chance": 40, "repeat": [ 1, 2 ] }, { "item": "farming_tools", "chance": 15 } ],
+        "4": { "item": "elec_parts", "chance": 25, "repeat": [ 1, 3 ] },
+        "5": { "item": "hardware_clothing", "chance": 25, "repeat": [ 1, 3 ] },
+        "6": [ { "item": "hand_tools", "chance": 25, "repeat": [ 1, 3 ] }, { "item": "power_tools", "chance": 15 } ],
+        "7": [ { "item": "farming_seeds", "chance": 40, "repeat": [ 1, 2 ] }, { "item": "farming_tools", "chance": 15 } ],
+        "8": { "item": "raw_materials", "chance": 35 },
+        "9": { "item": "raw_materials", "chance": 35 },
```
Comment (I-am-Erk) [inline data/json/mapgen/homeimprovement_superstore.json]: You could add many more repeats to all these store stock items, like 5-10. This isn't the sort of store that should start the game heavily looted.

File: data/json/mapgen/homeimprovement_superstore.json [hunk 3]
```diff
@@ -60,6 +60,11 @@
       [ "paint_brush", 55 ]
     ]
   },
+  {
+    "id": "carpet",
+    "type": "item_group",
+    "items": [ [ "r_carpet", 100 ], [ "p_carpet", 100 ], [ "y_carpet", 100 ], [ "g_carpet", 100 ], [ "p_carpet", 100 ] ]
+  },
```
Comment (I-am-Erk) [inline data/json/mapgen/homeimprovement_superstore.json]: Could you move all these item groups to the itemgroups folder, maybe to supplies.json? We should get away from defining item groups in map files, they get lost and nobody updates them

Comment (LovamKicsiGazsii) [reply data/json/mapgen/homeimprovement_superstore.json]: I'll try my best. I don't know if all of these item groups belong to ```supplies_json```.

**Comment by ralreegorganon** [comment]
You probably want to make this a new building rather than hijacking the `home_improvement_superstore` entry. Renaming the overmap terrains as is done here is going to cause errors when users load existing saves that have overmaps that already contain this.

The error-free way to add this is to add it as new content (new city_building, overmap_terrain, palette, etc) and update the entry in regional_map_settings, replacing the `home_improvement_superstore` with your new city_building id. 

The result will be that existing overmaps that include this building will generate without error, while new overmaps will use the new version rather than the old.

**Pull request merged**